examples/api/http/http.go

```diff
@@ -0,0 +1,115 @@
+package http
+
+import (
+	"context"
+	"fmt"
+	"github.com/Southclaws/fault"
+	"github.com/Southclaws/fault/fctx"
+	"github.com/Southclaws/fault/fmsg"
+	"github.com/Southclaws/fault/ftag"
+	"github.com/go-chi/chi/v5"
+	"github.com/go-chi/chi/v5/middleware"
+	"go.uber.org/zap"
+	"net/http"
+	"strings"
+	"time"
+)
+
+// Adding requestID in the fault context for logging purpose
+func RequestToFCtx(next http.Handler) http.Handler {
+	fn := func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+		rid := middleware.GetReqID(ctx)
+		ctx = fctx.WithMeta(ctx, "request_id", rid)
+		ctx = fctx.WithMeta(ctx, "http_method", r.Method)
+		ctx = fctx.WithMeta(ctx, "request_path", r.URL.Path)
+		ctx = fctx.WithMeta(ctx, "remote_ip", r.RemoteAddr)
+		ctx = fctx.WithMeta(ctx, "protocol", r.Proto)
+		next.ServeHTTP(w, r.WithContext(ctx))
+	}
+	return http.HandlerFunc(fn)
+}
+
+// Adding any path variable in the fault context for logging purpose
+func PathVariableAsFCtx(pathVarName, fctxName string) func(next http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		fn := func(w http.ResponseWriter, r *http.Request) {
+			if value := chi.URLParam(r, pathVarName); value != "" {
+				r = r.WithContext(fctx.WithMeta(r.Context(), fctxName, value))
+			}
+			next.ServeHTTP(w, r)
+		}
+		return http.HandlerFunc(fn)
+	}
+}
+
+func fctxToZapFields(ctx context.Context) []zap.Field {
```

Comment:
I actually think it would be good to use the stdlib slog here instead of zap - although zap is a popular choice for structured logging, perhaps slog is going to become the most common interface for structured logging and it would be nice to minimise specific third party dependencies in the example so that it shows how versatile Fault can be with any logging library.

But, happy to merge this now and update it later when slog is no longer experimental.

Reply:
That was good timing! https://go.dev/blog/go1.21

Reply:
I'll try to find the time to move to slog.